### PR TITLE
Document update

### PR DESCRIPTION
**Pull request by juhhei01**
Added documentation for RF driver extensions API.
Added documentation about SW MAC IEEE 802.15.4-2015 subset feature extension support.

### CODE REVIEW

**Comment by juhhei01** [comment]
@AnotherButler , @iriark01 , @mikter , @kjbracey-arm can you review my document Update.

**Comment by juhhei01** [comment]
@JarkkoPaso Please review this.

File: docs/reference/contributing/connectivity/MeshInterface.md [hunk 1]
```diff
@@ -170,6 +170,67 @@ State|Description
 
 ![tx](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/tx_process.png)
 
+### MESH PHY extended driver support
+
+Mesh IEEE 802.15.4 MAC will automatically use RF driver extended features. These features are needed when MAC user need to send IEEE 802.15.4 frames. MAC extensuon will enable if driver not support extended features.
```
Comment (mikter) [inline docs/reference/contributing/connectivity/MeshInterface.md]: Mesh IEEE 802.15.4 MAC will automatically use RF driver extended features if those are enabled in driver.

These features are needed when MAC user need to send IEEE 802.15.4 frames. -> this is not right?

MAC extensuon will enable if driver not support extended features.  -> MAC extensions are used automatically if driver implements the extension. If these extensions are not implemented the default behaviour is used thus limiting features.

File: docs/reference/contributing/connectivity/MeshInterface.md [hunk 2]
```diff
@@ -170,6 +170,67 @@ State|Description
 
 ![tx](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/tx_process.png)
 
+### MESH PHY extended driver support
+
+Mesh IEEE 802.15.4 MAC will automatically use RF driver extended features. These features are needed when MAC user need to send IEEE 802.15.4 frames. MAC extensuon will enable if driver not support extended features. 
+
+Extented driver features enable next feature's:
+
+- MAC to control packet TX time define CCA backoff period before CCA check
```
Comment (mikter) [inline docs/reference/contributing/connectivity/MeshInterface.md]: should the features be 
MAC controlled CCA/CSMA ?
support for IEEE 802.15.4-2015 enhanced ACK and information elements?

**Comment by mikter** [comment]
@karsev could you also review this

File: docs/reference/contributing/connectivity/MeshInterface.md (hunk 2 above)
Comment (juhhei01) [reply docs/reference/contributing/connectivity/MeshInterface.md]: OK

File: docs/reference/contributing/connectivity/MeshInterface.md (hunk 1 above)
Comment (juhhei01) [reply docs/reference/contributing/connectivity/MeshInterface.md]: ok

**Comment by AnotherButler** [comment]
ping @JarkkoPaso @kjbracey-arm @karsev

**Comment by juhhei01** [comment]
@AnotherButler, @mikter , @karsev, @kjbracey-arm  Hi Amanda this documentation will not be published in 5.10. We must verify implementation with real hardware and probably we will need update API. We have done already face to face review process about 80% of updates.

**Comment by AnotherButler** [comment]
@juhhei01 Thanks for letting me know 👍

**Comment by AnotherButler** [comment]
@juhhei01 Will this go into 5.11?

**Comment by juhhei01** [comment]
@AnotherButler Hi we need to discus with team how ready we are.

**Comment by AnotherButler** [comment]
Please feel free to reopen this when the team has determined it's ready to go in.

**Pull request closed without merging**